srv/modules/runners/orderednodes.py

```diff
@@ -0,0 +1,38 @@
+import salt.client
+
+
+def _preserve_order_sorted(seq):
+    """
+    Getting rid of duplicates in python could be solved by
+    casting a list() to a set() and back to a list()
+    `list(set(list_in_question))`
+    This method will mess with the sorting though.
+    As we rely on the sorting in this scenario, we have to use this
+    helper.
+    """
+    seen = set()
+    return [x for x in seq if x not in seen and not seen.add(x)]
+
+
+def unique(cluster=None, exclude=[]):
```

Comment:
cluster='ceph' will most likely be better as, you pointed out, None won't yield anything.

I'd still like to keep the varialbe with a default as it makes all the invocations easier to read -  especially where they appear most of the time - in jinja files.

`% for host in salt.saltutil.runner('orderednodes.unique', cluster='ceph') %}`

as opposed to

`% for host in salt.saltutil.runner('orderednodes.unique', 'ceph') %}`